packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2749,17 +2756,20 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
 
   @override
   void didChangeDependencies() {
-    // nullOk is valid here since  both the Scaffold and ScaffoldMessenger are
+    // Null is valid here since  both the Scaffold and ScaffoldMessenger are
     // currently available for managing SnackBars.
     final ScaffoldMessengerState? _currentScaffoldMessenger = ScaffoldMessenger.maybeOf(context);
     // If our ScaffoldMessenger has changed, unregister with the old one first.
     if (_scaffoldMessenger != null &&
       (_currentScaffoldMessenger == null || _scaffoldMessenger != _currentScaffoldMessenger)) {
       _scaffoldMessenger?._unregister(this);
     }
-    // Register with the current ScaffoldMessenger, if there is one.
-    _scaffoldMessenger = _currentScaffoldMessenger;
-    _scaffoldMessenger?._register(this);
+
+    if (widget.registerWithMessenger) {
```

Comment:
This runs every time any of the inherited widgets Scaffold depends on changes, so it seems we could end up registering with the same scaffold messenger more than once. Unless _register is safe to call more than once and ensures that the old registration is cleared before a new one is created?